README.md

```diff
@@ -1,6 +1,63 @@
+
+`scrape-negative` is used to collect examples for what is undesired by the censorer. It searches Google News with the comma-separated queries defined in the environment variable `NEGATIVE_QUERIES` in `server/.env`.\
+The scraped articles are then used in censoring as negative examples, where the [Word Mover's Distance](http://proceedings.mlr.press/v37/kusnerb15.pdf) of a paragraph to the scraped article's summaries plays a role in determining wether that paragraph should be censored.
+
+`scrape-positive` is used to collect examples for what is desired by the censorer. It, analogously to `scrape-negative`, searches Google News with the comma-separated queries defined in the environment variable `POSITIVE_QUERIES` in `server/.env`.\
+If `scrape-positive` is run with the argument `-t` or `--train`, the resulting articles from this scraping are used to train a [Biterm Topic Model](https://github.com/xiaohuiyan/xiaohuiyan.github.io/blob/master/paper/BTM-WWW13.pdf) with the parameters defined by the environment variables `TRAINING_*` as given in `server/.env`. Training is necessary on the first run, but can later be skipped to reuse the existing BT Model.\
+The Biterm Topic Model makes the key decision in finding which of the scraped positive, desired examples in the database will be used to replace a paragraph that is marked for censorship.
+
+Both `scrape-positive` as well as `scrape-negative` have an optional argument `-n` or `--narticles` that can be used to define an upper limit for how many arguments are scraped per query.
+
+Note that Google search operators such as the `site:` or `when:` modifiers  can strongly refine and empower defined search queries (e.g. `when:7d` constrains results to articles published in the past week). See [this incomplete list of operators](https://developers.google.com/search/docs/advanced/debug/search-operators/overview).
+
+##### Running the server
+
+With data collection and model training done, the server now has sufficient data to act as weBlock's backend. To run the backend, execute `server/run-backend`. Once it's ready, use the client as described above.
+
+## Scalability
+
+Since this is a proof-of-concept prototype and focussed on the natural language processing side of the project, some features that would be significant for scalability and real-world use have been left unattended for ease of use, ease of installation, project size and human resource prioritization. These include, but are not limited to
+
+- database & RAM: this project does not use a real database but instead simple text files and is strongly constrained by the RAM's size (e.g. the entire databases' contents may be loaded in RAM at times)
+- server: the current architecture uses a simple socket for serving the (*singular*) client
+- performance
```

Comment:
full performance optimization

(+ censoring of other media like images/ videos or documents)